FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Mutual_Auth_wolfSSL/README_wolfSSL.md

```diff
@@ -10,18 +10,112 @@ By double-clicking the solution file named "**mqtt_mutual_auth_demo_wolfSSL.sln*
+3. set client certificate
+4. set private key
+5. choose interface to use
  
+If even one of the above 1 to 4 is not set, an error will occur at build time. You should open **demo_config.h** to set them.
 
+<br>
 
-# Choose an interface to use
-You should choose an interface to use to configure the demo. However you may not know how to choose it. The demo will give you  good guidance. Follow the steps below.
+# Set Broker endpoint
 
-1. Build the RTOSDemo project
-2. Run the RTOSDemo.exe 
+A broker endpoint is a url that represents where MQTT subscribers and publishers access. In case your device is going to access AWS IoT device data endpoints, the endpoint would be the following format: ***account-specific-prefix*.iot.*aws-region*.amazonasw.com**.
+
+To set broker endpoint, find the statement '**#define democonfigMQTT_BROKER_ENDPOINT    "...insert here...**' in demo_config.h and activate it by copy & paste to the outside of the commented part. Replace "...insert here..." with your broker endpoint url.
+```
+#define democonfigMQTT_BROKER_ENDPOINT   "a****.iot.us-***.amazonaws.com"
+```
+You may find "democonfigMQTT_BROKER_PORT" just below of the democonfigMQTT_BROKER_ENDPOINT macro. If your MQTT broker port is "8883", no need to specifiy that value here, since the value is defined in MutualAuthMQTTExample.c by default.
+
+<br>
+
+# Set Credentials
+
+Since this demo handles mutual authentication, you need to provide rootCA certificate, client certificate and client private key. Those credentials should be set by way of following macros in demo_config.h:
+1. **democonfigROOT_CA_PEM**
+2. **democonfigCLIENT_CERTIFICATE_PEM**
+3. **democonfigCLIENT_PRIVATE_KEY_PEM**
+
+For setting those credentials, you have a option to specify the source of them, using file or using buffer. If you want provide credentials using buffer,
+activate **democonfigCREDENTIALS_IN_BUFFER** macro. Otherwise, let the macro commented out.
+```
+#define democonfigCREDENTIALS_IN_BUFFER
+```
+<br>
+
+## Setting credentials using file
+
+<br>
+
+By default, above **democonfigCREDENTIALS_IN_BUFFER** macro definition is commented out, therefore each credential should be provided by PEM encoded file. In this case, each macro definition looks like:
+
+```
+#define democonfigROOT_CA_PEM  "rootCA-PEM-encoded-file-path"
+```
+Activate two other macro definitions and set a file path for each.
+
+<br>
+
+## Setting credential using buffer
+
+<br>
+
+First of all, activate **democonfigCREDENTIALS_IN_BUFFER** macro.
+```
+#define democonfigCREDENTIALS_IN_BUFFER
+```
+Second, prepare a source file to contain the contents of the credentials. As an example, prepare credentials.c file and add it to the demo project. In the file, the contents of each credential need to be
+defined as "**const char***" variable. In this example, **root_ca**, **devide_cert** and **device_priv_key** represent buffers.
+
+```
+/* credentials.c */
+
+const char* root_ca =
+"-----BEGIN CERTIFICATE-----\n"

```

Comment:
Can you change these three strings to be `"Paste (object type) here.` for example `"Paste Root CA here."`

Reply:
Replaced them with suggested strings.